shell/platform/common/cpp/text_input_model.h

```diff
@@ -111,15 +160,39 @@ class TextInputModel {
   // The position of the cursor.
   int selection_extent() const { return selection_.extent(); }
 
+  // The position where the composing region starts.
+  int composing_base() const { return composing_range_.base(); }
+
+  // The position where the composing region ends.
+  int composing_extent() const { return composing_range_.extent(); }
```

Comment:
Are these four utility methods actually worth having now that you've added the range class?

Reply:
Will switch the Linux/Windows embedders over to use the new class.

Done in https://github.com/flutter/engine/pull/21854 for the selection bits in order to reduce unrelated noise in this patch for later archaeologists. Will rebase this on top of that.